src/models/trigger/cron.ts

```diff
@@ -0,0 +1,49 @@
+import * as avs_pb from "../../../grpc_codegen/avs_pb";
+import Trigger, { TriggerProps, TriggerTypes } from "./interface";
+
+// Required props for constructor: id, name, type and data: { scheduleList }
+export type CronTriggerDataType = avs_pb.CronCondition.AsObject;
+export type CronTriggerProps = TriggerProps & { data: CronTriggerDataType };
+
+class CronTrigger extends Trigger {
+  constructor(props: CronTriggerProps) {
+    super({ ...props, type: TriggerTypes.CRON, data: props.data });
+
+    console.log("CronTrigger.constructor.props:", {
+      ...props,
+      type: TriggerTypes.CRON,
+      data: props.data,
+    });
+  }
+
+  toRequest(): avs_pb.TaskTrigger {
+    const request = new avs_pb.TaskTrigger();
+    request.setName(this.name);
+
+    if (!this.data) {
+      throw new Error(`Trigger data is missing for ${this.type}`);
+    }
+
+    const condition = new avs_pb.CronCondition();
+    condition.setScheduleList((this.data as CronTriggerDataType).scheduleList);
+    request.setCron(condition);
+
+    console.log("CronTrigger.toRequest.request:", request.toObject());
+
+    return request;
+  }
+
+  static fromResponse(raw: avs_pb.TaskTrigger): CronTrigger {
+    // Convert the raw object to TriggerProps, which should keep name and id
+    const obj = raw.toObject() as unknown as TriggerProps;
+
+    console.log("CronTrigger.fromResponse.obj:", obj);
+    return new CronTrigger({
+      ...obj,
+      type: TriggerTypes.CRON,
```

Comment:
hey @chrisli30 these `type` parameter isn't needed when sending to protobuf. we can remove them.

Reply:
Understood, but these `type` is not sent to AVS currently, but as an intrinsic variable of the SDK’s Trigger. 

Each Trigger has their own data struct, without a shared `type` variable, it‘s hard to use shared code, in other words, class inheritance.

Reply:
we can juse use sth as above with `.getType()` method define in the parent Trigger class to get class name and parse out the type.

Reply:
Yes, that will work. However, I’m trying to achieve something extra: to directly reference the code you defined in the original protobuf, which is the below code.

```
export namespace TaskTrigger {
    export enum TriggerTypeCase {
        TRIGGER_TYPE_NOT_SET = 0,
        MANUAL = 2,
        FIXED_TIME = 3,
        CRON = 4,
        BLOCK = 5,
        EVENT = 6,
    }
}
```

Even if we getType() the classNames, they are defined by the sdk. If we change anything in the protobuf, the classNames will need manual update as well.